packages/amplify-category-analytics/tests/commands/add.test.js

```diff
@@ -0,0 +1,47 @@
+const add = require('../../commands/analytics/add');
+
+describe('analytics add: ', () => {
+  const mockExecuteProviderUtils = jest.fn();
+  const mockGetProjectDetails = jest.fn();
+  const mockSelectionPrompt = jest.fn(() => Promise.resolve({ service: 'Pinpoint', providerName: 'awscloudformation' }));
+  const mockSelectionPromptWithError = jest.fn(() => Promise.reject('Error'));
+
+  const mockContext = {
+    amplify: {
+      executeProviderUtils: mockExecuteProviderUtils,
+      getProjectDetails: mockGetProjectDetails,
+      serviceSelectionPrompt: mockSelectionPrompt,
+      updateamplifyMetaAfterResourceAdd: jest.fn(),
+    },
+    print: {
+      warning: jest.fn().mockImplementation(info => console.log(info)),
+      info: jest.fn().mockImplementation(info => console.log(info)),
+      error: jest.fn().mockImplementation(info => console.log(info)),
+      success: jest.fn().mockImplementation(info => console.log(info)),
+    },
+  };
+
+
+  it('unsuccessful analytics add with Cloudformation provider', async () => {
+    jest.mock('../../provider-utils/awscloudformation/index', () => undefined);
+    await add.run(mockContext);
+    expect(mockContext.print.error).toBeCalledWith('Provider not configured for this category');
+  });
+
+
+  it('successful analytics add with Cloudformation provider', async () => {
+    jest.mock('../../provider-utils/awscloudformation/index', () => ({ addResource: () => Promise.resolve('analyticsresource') }));
+    await add.run(mockContext);
+    expect(add.run).toBeDefined();
```

Comment:
I don't think we need this assertion. If this is missing tests will fail. If you insist on having this move it to line 34